index/scorch/merge.go

```diff
@@ -129,8 +147,79 @@ func (s *Scorch) parseMergePlannerOptions() (*mergeplan.MergePlanOptions,
 	return &mergePlannerOptions, nil
 }
 
+type mergerCtrl struct {
+	creator string
+	options *mergeplan.MergePlanOptions
+	doneCh  chan struct{}
+}
+
+func (s *Scorch) RequestMerge(mo *mergeplan.MergePlanOptions,
+	doneCh chan struct{}) error {
+	// check whether force merge is already under processing
+	if atomic.LoadUint64(&s.stats.TotFileMergeForceOpsStarted) >
+		atomic.LoadUint64(&s.stats.TotFileMergeForceOpsCompleted) {
+		if doneCh != nil {
+			close(doneCh)
+		}
+		return fmt.Errorf("compaction already in progress")
```

Comment:
maybe just minor naming thing here, the API operation is called merge, but the error message talks about compaction